### ISSUE TITLE
Npm Package

### ISSUE THREAD

**Issue by yargalot** [Odecee/angular-form-lib#1]
Hey,

Any chance we can set this up so we can consume this via a NPM package?

Thanks :)

**Comment by uglow** [comment]
I think so. Was there any specific NPM setup you require, or is adding this library to the NPM registry sufficient?

**Comment by yargalot** [comment]
Well, if we can add 
```/* commonjs package manager support (eg componentjs) */
if (typeof module !== "undefined" && typeof exports !== "undefined" && module.exports === exports){
  module.exports = 'angular-form-lib';
}```

and then point to the main release file in the package.json
```
"main": "lib/ng-form-lib.js",
```

I think that should be all to require them in the commonjs projects hopefully?

**Comment by uglow** [comment]
If I use `grunt-umd`, it produces this wrapper around the code:
```
(function (root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module unless amdModuleId is set
    define([], function () {
      return (root['angular-form-lib'] = factory());
    });
  } else if (typeof exports === 'object') {
    // Node. Does not work with strict CommonJS, but
    // only CommonJS-like environments that support module.exports,
    // like Node.
    module.exports = factory();
  } else {
    root['angular-form-lib'] = factory();
  }
}(this, function () {

/*
 Copyright 2014-2015 angular-form-lib project contributors (see CONTRIBUTORS.md).
 Licenced under Apache 2.0 licence (see LICENCE.txt)
*/

(function(angular) {
  'use strict';
  ...
})(window.angular);

return angular-form-lib;

}));
```

That last return statement looks like it wouldn't even compile!

Alternatively, if all you need is:
```
/* commonjs package manager support (eg componentjs) */
if (typeof module !== "undefined" && typeof exports !== "undefined" && module.exports === exports){
  module.exports = 'angular-form-lib';
}
```
...then I can add that easily enough.

**Comment by uglow** [comment]
Added commonjs code as part of 8c6794f047beb91d6e06d9f0150cb275f538e0c1, tagged as `v0.4.0`. @yargalot, please you confirm this meets your needs. If I don't hear anything in 5 days, I will assume this issue is closed.

**Comment by yargalot** [comment]
Righto, will give it a test this afternoon ;)

**Comment by yargalot** [comment]
Dumb question, Do you want me to reference github directly to test or pull down from NPM? Not sure if you have published it or not :)

**Comment by uglow** [comment]
The package is now registered with NPM - please go ahead :+1:

**Comment by yargalot** [comment]
Hey Brett, I screwed up, Messed up the exports name, should be ngFormLIb, other than that, it all worked perfectly.

**Comment by uglow** [comment]
Code merged into v0.4.3. Thanks Steven!
